go-controller/pkg/ovn/egressgw.go

```diff
@@ -256,25 +256,34 @@ func (oc *Controller) addGWRoutesForPod(routingGWs []net.IP, podIfAddrs []*net.I
 	}
 	defer nsInfo.Unlock()
 	gr := "GR_" + node
-	for _, v := range routingGWs {
-		gw := v.String()
-		// TODO (trozet): use the go bindings here and batch commands
-		for _, podIPNet := range podIfAddrs {
-			podIP := podIPNet.IP.String()
-			mask := GetIPFullMask(podIP)
-			_, stderr, err := util.RunOVNNbctl("--may-exist", "--policy=src-ip", "--ecmp-symmetric-reply",
-				"lr-route-add", gr, podIP+mask, gw)
-			if err != nil {
-				return fmt.Errorf("unable to add external gw src-ip route to GR router, stderr:%q, err:%v", stderr, err)
-			}
-
-			if err := oc.addHybridRoutePolicyForPod(podIPNet.IP, node); err != nil {
-				return err
-			}
-			if nsInfo.podExternalRoutes[podIP] == nil {
-				nsInfo.podExternalRoutes[podIP] = make(map[string]string)
+	for _, podIPNet := range podIfAddrs {
```

Comment:
A suggestion: it feels like this could be done more "ovn-kubernetes-y". We already have a method `util.MatchIPFamily` which does what you need. 

So we could do something like:

```go
for _, podIPNet := range podIfAddrs {
      if gw, err := util.MatchIPFamily(utilnet.IsIPv6(podIPNet), routingGWs); if err == nil {
           ...function body
      }
}
```
As such we can remove `ipFamilyMatch`

Reply:
awesome, thanks @alexanderConstantinescu I hadn't seen that.

Reply:
CI is failing after changing to `MatchIPFamily` Dan suggesting @danwinship suggestion I changing `MatchIPFamily` to return an array and changing changing the usages to `gwStr := gw[0].String()` but it's still failing CI. Still trying to figure out how to get CI passing

Below is what I tried modifying `MatchIPFamily` to:
```
func MatchIPFamily(isIPv6 bool, ips []net.IP) ([]net.IP, error) {
	var ipAddrs []net.IP
	for _, ip := range ips {
		if utilnet.IsIPv6(ip) == isIPv6 {
			ipAddrs = append(ips, ip)
		}
	}
	if len(ipAddrs) > 0 {
		return ipAddrs, nil
	}
	return nil, fmt.Errorf("no %s IP available", IPFamilyName(isIPv6))
}
```
Any ideas welcome! Ty!

Reply:
Yes, you're probably not supposed to be doing `gwStr := gw[0].String()` in that case, but instead loop over each. 

So, if `MatchIPFamily` now returns `[]net.IP`

```go
for _, podIPNet := range podIfAddrs {
      if gws, err := util.MatchIPFamily(utilnet.IsIPv6(podIPNet), routingGWs); if err == nil {
           for _, gw := range gws {
               // do the rest
           } 
      }
}
```

Reply:
Done and thanks for the assist guys!